src/github.com/getlantern/flashlight/config/config.go

```diff
@@ -115,9 +113,17 @@ func Init() (*Config, error) {
 			return
 		},
 	}
-	initial, err := m.Start()
+	initial, err := m.Init()
 	var cfg *Config
 	if err == nil {
+		er := pubsub.Sub(&geo.City{}, func(city *geo.City) {
+			log.Debugf("Got location: %v", city.Country.IsoCode)
+			m.StartPolling()
```

Comment:
Why start a new polling goroutine whenever our location changes?  I see you added a mutex in yamlconf to protect against some of the damage, but this still builds up unnecessary polling requests.

This could be prevented by creating a `sync.Once` out of this callback (see above) and changing this line to
```go
once.Do(func () { m.StartPolling() })
```